libraries/commons/ui/lib-helper-overlay/src/lib/components/OverlayTourProvider.tsx

```diff
@@ -7,67 +13,99 @@ import {
 } from "../..";
 
 export const OverlayTourContext = createContext<OverlayTourContextType>({
-  configs: [],
   addConfig: (arg) => {},
   removeConfig: (arg) => {},
   showSecondaryWithKey: null,
   setSecondaryWithKey: (key) => {},
-  showOverlay: (show) => {},
+  showOverlayHandler: () => {},
```

Comment:
@PavelOlkhovoi made these handlers explicit and added some memoization and useCallbacks to reduce rerenders.